doc/tunnel/SAI-Proposal-EVPN-Multihoming.md

```diff
@@ -0,0 +1,616 @@
+     * When set to true, egress unicast traffic will be dropped
+     *  
+     * Valid only when the SAI_VLAN_MEMBER_ATTR_BRIDGE_PORT_ID is of type PORT.
+     * 
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     * 
+     */
+    SAI_VLAN_MEMBER_ATTR_TX_DROP,
+
+```
+
+
+# 4.0 Sample Workflow
+
+This section describes the SAI object usage for different EVPN MH scenarios.
+
+
+## 4.1 Known Unicast workflow
+
+![EVPN Multihoming](figures/sai_evpnmh_unicast.png "Figure 1: Known Unicast Packet Flow")
+__Figure 1: Known Unicast Packet Flow__
+
+At VTEP5 the following objects are created.
+
+  - tnl_oid_1-tnl_oid_4 of type SAI_OBJECT_TYPE_TUNNEL corresponding to tunnels created to VTEP1-VTEP4.
+
+```
+    sai_attribute_t attr;
+    std::vector<sai_attribute_t> tunnel_attrs;
+
+    attr.id = SAI_TUNNEL_ATTR_TYPE;
+    attr.value.s32 = SAI_TUNNEL_TYPE_VXLAN;
+    tunnel_attrs.push_back(attr);
+
+    attr.id = SAI_TUNNEL_ATTR_PEER_MODE;
+    attr.value.s32 = SAI_TUNNEL_PEER_MODE_P2P;
+    tunnel_attrs.push_back(attr);
+
+    attr.id = SAI_TUNNEL_ATTR_ENCAP_DST_IP;
+    attr.value.ipaddr = vtep1_ip; 
+    tunnel_attrs.push_back(attr);
+
+    sai_status_t status = sai_tunnel_api->create_tunnel(
+                                &tnl_oid_1,
+                                gSwitchId,
+                                static_cast<uint32_t>(tunnel_attrs.size()),
+                                tunnel_attrs.data()
+                          );
+
+    /* create_tunnel for vtep2_ip to vtep4_ip */
+    ..................
+    ..................
+
+```
+  
+  - nh_oid_1 to nh_oid_4 of type SAI_OBJECT_TYPE_NEXT_HOP corresponding to tunnels (tnl_oid_1-tnl_oid_4)
+
+```
+    std::vector<sai_attribute_t> next_hop_attrs;
+    sai_attribute_t next_hop_attr;
+
+    next_hop_attr.id = SAI_NEXT_HOP_ATTR_TYPE;
+    next_hop_attr.value.s32 = SAI_NEXT_HOP_TYPE_BRIDGE_PORT;
+    next_hop_attrs.push_back(next_hop_attr);
+
+    next_hop_attr.id = SAI_NEXT_HOP_ATTR_IP;
+    next_hop_attr.value.ipaddr = vtep1_ip;
+    next_hop_attrs.push_back(next_hop_attr);
+
+    next_hop_attr.id = SAI_NEXT_HOP_ATTR_TUNNEL_ID;
+    next_hop_attr.value.oid = tnl_oid_1;
+    next_hop_attrs.push_back(next_hop_attr);
+
+    sai_status_t status = sai_next_hop_api->create_next_hop(&nh_oid_1, gSwitchId,
+                                            static_cast<uint32_t>(next_hop_attrs.size()),
+                                            next_hop_attrs.data());
+
+    /* create_next_hop for vtep2_ip/tnl_oid_2 to vtep4_ip/tnl_oid_4 */
+    ..................
+    ..................
+
+```
+
+
+  - nh_grp_oid_1 of type SAI_OBJECT_TYPE_NEXT_HOP_GROUP for the remote VTEP group (VTEP1-4)
+
+```
+    sai_attribute_t nhg_attr;
+    vector<sai_attribute_t> nhg_attrs;
+
+    nhg_attr.id = SAI_NEXT_HOP_GROUP_ATTR_TYPE;
+    nhg_attr.value.s32 = SAI_NEXT_HOP_GROUP_TYPE_BRIDGE_PORT;
+    nhg_attrs.push_back(nhg_attr);
+
+    sai_object_id_t next_hop_group_id;

```

Comment:
we shall use "sai_object_id_t nh_grp_oid_1"

Reply:
Ack